engine/runtime/src/main/scala/org/enso/interpreter/runtime/RuntimeStubsGenerator.scala

```diff
@@ -48,11 +48,25 @@ class RuntimeStubsGenerator(builtins: Builtins) {
         scope.registerType(builtinType.getType)
         builtinType.getType.setShadowDefinitions(scope, true)
       } else {
+        val isTypeProjectPrivate = tp.members.forall(_.isProjectPrivate)
```

Comment:
Thank you. Immutable `Type.isProjectPrivate` is better for reasoning about the overall behavior of the system.